week-5/endpoints.ts

```diff
@@ -0,0 +1,106 @@
+import * as url from 'url';
+import * as http from 'http';
+import UserModel from './UserModel';
+import type { HobbiesResponse, PostUser, UserResponse, SuccessfulResponse } from './types';
+
+const domain = "http://localhost:8000";
+
+export const processGetRequest = (userModel: UserModel, req: http.IncomingMessage): HobbiesResponse | undefined | UserResponse[] => {
+  const url = req.url;
+  if(url === undefined ) {
+    return undefined;
```

Comment:
You don't need to explicitly return undefined, using the return with no value is automatically assumed as undefined
```suggestion
    return;
```